packages/engine-core/src/binary/BinaryEngine.ts

```diff
@@ -895,16 +891,25 @@ You very likely have the wrong "binaryTarget" defined in the schema.prisma file.
 
   async request<T>(
     query: EngineQuery,
-    { headers = {}, numTry = 1, isWrite, transaction }: RequestOptions<undefined>,
+    { traceparent, numTry = 1, isWrite, transaction, debugPanic }: RequestOptions<undefined>,
   ): Promise<QueryEngineResult<T>> {
     await this.start()
+    const headers: Record<string, string> = {}
+    if (traceparent) {
+      headers.traceparent = traceparent
+    }
 
-    // TODO: we don't need the transactionId "runtime header" anymore, we can use the txInfo object here
-    this.currentRequestPromise = this.connection.post(
-      '/',
-      stringifyQuery(query.query),
-      runtimeHeadersToHttpHeaders(headers),
-    )
+    if (transaction) {
+      headers['X-transaction-id'] = transaction.id
```

Comment:
I think this is currently sending a batch id as a header, is it correct?

Reply:
No, batch ids are not propagated to `engine-core`. I'll rename paramter to make it clearer.

Reply:
it's now named `interactiveTransaction`.